### ISSUE TITLE
Tree-based FastKick?

### ISSUE THREAD

**Issue by rieder** [amusecode/amuse#669]
FastKick is very useful in Bridged simulations, but it scales with N1*N2, making it still a bottleneck in simulations with many particles (especially where N1 ≈ N2).

Since codes like gadget and phantom don't (yet) support get_gravity_at_point directly, it would be helpful to have a tree-based slim kicker code (preferably OpenMP-enabled) for such simulations. This would scale with log(N1)*N2.

**Comment by ipelupessy** [comment]
Fi is tree based ad openmp parallel, but I agree that another code might be nice...bonsai on GPU would be nice...

**Comment by ipelupessy** [comment]
bonsai currently has a host implementation of get_Gravity_at_point...so thats not so nice...maybe something like adding ghost particles with zero mass and then doing a a single gravity evaluation (and removing zero mass parts again) would work..

**Comment by ipelupessy** [comment]
I am pretty sure @jbedorf should be able to hack something in quickly ;-)

**Comment by rieder** [comment]
Yes I used Fi for this before, but it seems to complain about a “too short timestep” sometimes... which is weird for a kick-only code, since it doesn’t require a timestep.

**Comment by ipelupessy** [comment]
yea, it initializes always, necessitating a timestep shift (maybe this could be turned off)

**Comment by rieder** [comment]
To summarise:
- A bug in Bridge was causing codes to do internal force calculations, which prevented them from calculating just the kick
- Fi can indeed do this and is now suitable for my purpose
- It would still be nice to have a dedicated "TreeKick" code (so I'm keeping this issue open)
